libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -1050,6 +1071,34 @@ void AP_Baro::handle_external(const AP_ExternalAHRS::baro_data_message_t &pkt)
 }
 #endif  // AP_BARO_EXTERNALAHRS_ENABLED
 
+// returns false if we fail arming checks, in which case the buffer will be populated with a failure message
+bool AP_Baro::arming_checks(size_t buflen, char *buffer) const
+{
+    if (!all_healthy()) {
+        hal.util->snprintf(buffer, buflen, "Barometer not healthy");
+        return false;
+    }
+
+#if APM_BUILD_COPTER_OR_HELI || APM_BUILD_TYPE(APM_BUILD_ArduPlane)
```

Comment:
If the check only works on copter and plane we should only expose the param on copter and plane.

Reply:
That's problematic as there's a header involved.  I'd be inclined to leave it in on all vehicles except AP_Periph (but have an `AP_BARO_GPS_CHECKS_ENABLED` to allow it to be disabled).

Tracker might start feeding bad data up to a vehicle if we limit this check to Copter/Plane.

Reply:
The variable can still be defined in the header, we should just remove it from the table.

Reply:
yep, done